### PR TITLE
Add AndAlso and OrElse tests

### PR DESCRIPTION
**Pull request by hughbe**
- Without method
- With method
- Semantics (e.g. does AndAlso short-circuit)
- Invalid operator

Also increases CC by ~1.5%, contributing to #1198
And removes some dead code from VerifyOpTrueFalse

/cc @VSadov

### CODE REVIEW

File: src/System.Linq.Expressions/src/System/Linq/Expressions/BinaryExpression.cs [hunk 1]
```diff
@@ -893,8 +893,7 @@ private static void ValidateUserDefinedConditionalLogicOperator(ExpressionType n
         private static void VerifyOpTrueFalse(ExpressionType nodeType, Type left, MethodInfo opTrue, string paramName)
         {
             ParameterInfo[] pmsOpTrue = opTrue.GetParametersCached();
-            if (pmsOpTrue.Length != 1)
-                throw Error.IncorrectNumberOfMethodCallArguments(opTrue, paramName);
+            Debug.Assert(pmsOpTrue.Length == 1);
```
Comment (hughbe) [inline src/System.Linq.Expressions/src/System/Linq/Expressions/BinaryExpression.cs]: The method is retrieved by calling `TypeUtils.GetBooleanOperator(Type, string)`, found here: https://github.com/hughbe/corefx/blob/a82ade74aa16817e80e6df0bafaab47cbecd3204/src/System.Linq.Expressions/src/System/Dynamic/Utils/TypeUtils.cs#L703

This method calls `type.GetAnyStaticMethodValidated(name, new Type[] { type });`. This means that we will always only have one parameter!

**Comment by hughbe** [comment]
Test Innerloop OSX Debug Build and Test

File: src/System.Linq.Expressions/src/System/Linq/Expressions/BinaryExpression.cs (hunk 1 above)
Comment (stephentoub) [inline src/System.Linq.Expressions/src/System/Linq/Expressions/BinaryExpression.cs]: I realize it's dead code, but I think this should be reverted.  The entire purpose of this VerifyOpTrueFalse method is to verify and throw if the provided member is not true/false, and that includes validating the number of arguments.

**Comment by stephentoub** [comment]
Thanks, Hugh.  Please remove the first commit; otherwise, the updated tests look good.

**Comment by hughbe** [comment]
Sure, I removed the dead code commit

**Comment by stephentoub** [comment]
@dotnet-bot Test Innerloop CentOS7.1 Release Build and Test please ("failed to mkdirs")

**Pull request merged**